### DISCUSSION TITLE
Escaping the VoID

### DISCUSSION THREAD

**Discussion by briesenberg07** [uwlib-cams/uwlswd_datasets#24] [General]
# Related issues
- [x] https://github.com/uwlib-cams/uwlswd/issues/43
- [ ] #44 (??? can't find any issue 44)

# Background

We published most of the following datasets back in 2018. The distinct datasets were organized by class.

| IRI | dct:title |
|---|---|
| https://doi.org/10.6069/uwlib.55.A.3.1 | Instances of the SourceResource Class |
| https://doi.org/10.6069/uwlib.55.A.3.2 | Instances of the Aggregation Class |
| https://doi.org/10.6069/uwlib.55.a.3.3 | Instances of the WebResource Class |
| https://doi.org/10.6069/uwlib.55.a.3.4 | Instances of the Collection Class |
| https://doi.org/10.6069/uwlib.55.A.3.5 | Instances of the RightsStatement Class |
| https://doi.org/10.6069/uwlib.55.A.3.6 | Instances of the Agent Class |
| https://doi.org/10.6069/uwlswd.2e2b-y833 | Instances of the ProvenanceStatement Class |

Each of these datasets contains only a *very minimal* self-description. As of writing, only the following for the dataset resource (DOIs as given in table above):

```
<> rdf:type void:Dataset ;
	void:inDataset <https://doi.org/10.6069/uwlib.55.a#uwSemWeb> .
```

More complete descriptions for each are included as triples in the following resource...

| IRI | dct:title |
|---|---|
| https://doi.org/10.6069/uwlib.55.a | VoID Description of the dataset 'University of Washington Libraries' Semantic Web Data' | 

...where each of the "Instances of [...]" datasets are linked to a dataset resource `https://doi.org/10.6069/uwlib.55.a#uwSemWeb` in the following way:

```
<https://doi.org/10.6069/uwlib.55.a> foaf:primaryTopic <https://doi.org/10.6069/uwlib.55.a#uwSemWeb> .
<https://doi.org/10.6069/uwlib.55.a#uwSemWeb> void:classPartition <https://doi.org/10.6069/uwlib.55.a.3.1> , 
	<https://doi.org/10.6069/uwlib.55.a.3.2> , 
	<https://doi.org/10.6069/uwlib.55.a.3.3> , 
	<https://doi.org/10.6069/uwlib.55.a.3.4> , 
	<https://doi.org/10.6069/uwlib.55.a.3.5> , 
	<https://doi.org/10.6069/uwlib.55.a.3.6> , 
	<https://doi.org/10.6069/uwlswd.2e2b-y833> .
```

We would like to move description for each of the uwSemWeb class partitions ("Instances of [...]" datasets) into those datasets so that -- at least, from the perspective of someone who downloads, for example, the file [instances_of_the_provenancestatement_class.ttl](https://uwlib-cams.github.io/uwlswd_datasets/instances_of_the_provenancestatement_class.ttl) -- they self-describe.

# 1. What to require for "Instances of [...]" dataset self-description?
Perhaps we've already answered this! The #uwSemWeb dataset contains triples describing each "Instance of [...]" partition which we may wish to simply move into each such dataset to create (at least part of) the desired self-description. These triples include:

```
<> void:Class <iri> ;
	dct:title "literal"@en ;
	owl:version "literal"@zxx ;
	dct:issued "YYYY"@zxx ;
	dct:hasFormat <url>, <url>, <url>, <url> .
        dct:description: "literal"@en ;
```

Subjects of these triples are "Instances of [...]" dataset IRIs. Subject IRIs would not need to change. Triples would simply need to be moved into different *files*.

Also to consider: Align self-description of these datasets with self-description for the UWLSWD vocabularies.

# 2. Whether/how to maintain/deprecate the `uwlib.55.a` dataset description resource and the uwlib.55.a#uwSemWeb dataset resource
...

**Comment by gerontakos** [comment]
Although I'm not saying anything is "wrong" above, I don't concur with the way the problem is framed. A dataset description, in our data, is not "in" the data. Additionally, we're crossing into dubious ground if we call our dataset descriptions "self description" (in the W3C TAG meaning of the term). I raise this issue again because I think it's significant. Being cautious with the way we think about the data as data (not just as pawns in our workflow, or as files retrieved by a person downloading) might help with long-term vision; then again, it might not...

I think the problem is that we no longer want to claim that our entity-based datasets are part of a single dataset. We set it up that way becuase we intended it to be the "DPLA" dataset (actually that's not true; it turned into that; originally i believe ALL UW RDF data was supposed to be a subset of uwlib.55.a#uwSemWeb). It seemed like a good idea, but now we're growing, and either we have to keep a separate "DPLA" dataset, or allow it to integrate, kind-of, with other entity-based datasets. We're choosing the latter. In that context, it's difficult to justify the continuation of the dataset uwlib.55.a#uwSemWeb. One reason is that it did not provide helpful information; I would say it may have even made our data collection more confusing. But what's lost when we do this? I would say that the DPLA data is more difficult to assemble; the only thing holding the "DPLA" data together are the triples describing the Aggregations. But that's infoSci life, no? Source resources are SRs and are not the sole possession of "DPLA data." They become a CHO when an aggregation is asserted. And so, all is as it should be, no? 

So we want to deprecate the triples with subjects <https://doi.org/10.6069/uwlib.55.a> and <https://doi.org/10.6069/uwlib.55.a#uwSemWeb>. We want to maintain selected triples, however, with subjects represented by the IRIs of our entity-based datasets: <https://doi.org/10.6069/uwlib.55.a.3.1>, <https://doi.org/10.6069/uwlib.55.a.3.2>, etc. So, in our prehistoric way of creating and disseminating data, we can either leave those triples in the current file or move them to same file in which we store the datasets those triples describe. In the world of data, it's exactly the same thing; in our world, the Data Dump world, it's not the same thing. For us, our workflows are eased if we throw dataset descriptions in the same file as the datasets themselves. For our data consumers, we think it would also help them, as we force them to Dumpster-Dive to find the triples they need. So it's a hypothetical win-win! 

So here's what I think we need to decide (as suggested by Benjamin already):
(1) What are the requirements for a minimal descriptions of a dataset (or, if it's different, an entity-based dataset) at UW Libraries? When we decide that, we may be able to use some of the old triples. We'll probably need additional triples too.
(2) How do we deprecate <https://doi.org/10.6069/uwlib.55.a#uwSemWeb>  and <https://doi.org/10.6069/uwlib.55.a> ? I suspect we just leave it in place and strip-out false or redundant triples. Then those resources would not be incorrect (unless we add to the "partitions" in ways that contradict the assertions in uwlib.55.a and uwlib.55.a#uwSemWeb), so who are they hurting? Maybe us, as we'd have to manage the DOI, but I bet we can live with that.

This all also raises the question of why we're separating datasets at all -- that's another discussion. Preview: why would we ever create a "dataset" with a distinct IRI/DOI if it is not intended to be a named graph in the future?

**Reply by briesenberg07** [reply]
Thank you @gerontakos !  Everyone's favorite, 'responses to responses':

> **A dataset description, in our data, is not "in" the data. Additionally, we're crossing into dubious ground if we call our dataset descriptions "self description" (in the W3C TAG meaning of the term).**

I think I need further discussion in order to understand this, especially if it seems significant to you. When you say 'dataset description' are you  referring to triples with subject https://doi.org/10.6069/uwlib.55.a , which is typed as void:DatasetDescription? Or to triples in various of our vocabularies which have as subject a DOI with no fragment identifier, such as for example:

```
<https://doi.org/10.6069/uwlib.55.d.4>
  a void:Dataset ;
  dct:title "University of Washington Libraries RDA Application Profile Extension"@en ;
  dct:description "RDF dataset describing properties required beyond those provided by the RDA Registry to describe RDA Entities at University of Washington Libraries."@en ;
  dct:creator <http://viaf.org/viaf/139541794> ;
  dct:publisher <http://viaf.org/viaf/139541794> ;
  dct:issued "2019"^^xsd:gYear ;
  dct:language <http://id.loc.gov/vocabulary/languages/eng> ;
  dct:license <http://creativecommons.org/publicdomain/zero/1.0/> 
  [...]
```
... or to something else?

> **[...] we no longer want to claim that our entity-based datasets are part of a single dataset [...]**

Confirmed ✔️ agree 👍🏻 

> **[...] we can either leave those triples in the current file or move them to same file in which we store the datasets those triples describe. [...] For us, our workflows are eased if we throw dataset descriptions in the same file as the datasets themselves. For our data consumers, we think it would also help them, as we force them to Dumpster-Dive to find the triples they need. So it's a hypothetical win-win!**

Confirmed ✔️ agree 👍🏻 

> **(2) How do we deprecate https://doi.org/10.6069/uwlib.55.a#uwSemWeb and https://doi.org/10.6069/uwlib.55.a ? I suspect we just leave it in place and strip-out false or redundant triples. Then those resources would not be incorrect (unless we add to the "partitions" in ways that contradict the assertions in uwlib.55.a and uwlib.55.a#uwSemWeb), so who are they hurting? Maybe us, as we'd have to manage the DOI, but I bet we can live with that.**

~I feel a new issue coming on... For hashing out the details?~ See #44 

> **I would say that the DPLA data is more difficult to assemble; the only thing holding the "DPLA" data together are the triples describing the Aggregations. But that's infoSci life, no? Source resources are SRs and are not the sole possession of "DPLA data." They become a CHO when an aggregation is asserted. And so, all is as it should be, no?**

For the future... 🏜️🏝️🗻... I love our goal of publishing our collections-as-linked-open-data, but I'm quite down on the DPLA MAP. I'd like to find another, better ontology/ies to use in future (ones that don't dereference to a PDF...), and take necessary steps to deprecate the datasets which are instances of DPLA classes... but this may be controversial, so it's for *future*

**Comment by gerontakos** [comment]
"dataset descriptions." All of them; both the triples typed as void:DatasetDescription and the ones in the same files as the datasets. Neither is "in" the data. Or so I say.

You know what, I don't even know what a dataset is. It could very well be a file, for all I know.

All I'm seeing is N-Triples/N-Quads. I know we often talk of graphs: sets of triples. The fourth element of a quad is often the graph name, right? What I've heard before, I don't recall where, is that the "dataset" is one or more graphs that follow a common data model or shape. No matter how you slice it, however, I don't see any way to say that data and metadata are part of the same dataset.

This is easier to explore using speech. It's easy to overthink this. But I think, in our data dumpster, what makes most sense to me, and, hopefully, to consumers, is the approach we took for our newer datasets (not the "DPLA dataset" typed void:DatasetDescription): 
https://doi.org/10.6069/uwlib.55.d.3 (for example) dereferences to a full dataset (a "web resource").
What appears first (if we can talk about anything being "first") is often the "useful information" (TBL's rule 3 at that horrible web page https://www.w3.org/DesignIssues/LinkedData.html) associated with 55.a.3.1; that's metadata, but it has no identity, it describes the dataset and you retrieve that metadata when you request the dataset by IRI. Weird! 
Then what follows are the secondary resources, usually RWOs, identified by hash IRIs like https://doi.org/10.6069/uwlib.55.d.3#inDigiColl. In this particular example, 55.d.3, the secondary resources are not RWOs but web documents; the "dataset" IRI retrieves a "web document" as expected, but "contains" another "web document" as a secondary resource identified by a hash identifier. No prohibition against that! So it works. And it makes sense to me. I just want to do that over and over without thinking much more about it -- which includes just getting rid of (or deprecating) https://doi.org/10.6069/uwlib.55.a.

**Reply by briesenberg07** [reply]
Thank you @gerontakos . I think this more or less matches my mental model of the difference between the UWLSWD IRIs which *don't include* a fragment identifier, and those that *do*. So, it's nice to (I think) confirm that we are thinking about these in the same way.

https://doi.org/10.6069/uwlib.55.A.3.6 = a Dataset = something that can be transmitted over HTTP
https://doi.org/10.6069/uwlib.55.A.3.6#CurtisAsahel18741941 = an agent = something that can't be transmitted over HTTP

I have a vague sense that these two categories (in my mental model) are at least reasonably in line with linked-data best practices (I think you have a much better sense of whether or not this is the case than I do).

**Comment by gerontakos** [comment]
Yes, we'll want to work out the details of UW Libraries deprecation.

**Reply by briesenberg07** [reply]
Created #44 as a place for this

**Comment by gerontakos** [comment]
The DPLA ontology is pretty good. We could jump ship to Europeana, which the DPLA MAP is based on. Then we could use RDA itself. Or CIDOC-CRM!! But were we to go forward with cdm-objects-expressed-as-DPLA-entities, I think using the DPLA MAP is fine. First, there's tons of resources that actually use this data model (in the DPLA itself); then, if we're terribly concerned that it is not expressed in RDF, we could do it ourselves. In fact, that would be a good project. I don't think it would be too difficult. It's a well thought-out MAP. A small ontology would cover all the unique properties/classes, and then a shape to cover the MAP. FOR THE FUTURE!

**Reply by briesenberg07** [reply]
For the future... 🏜️🏝️🗻!

**Comment by cspayne** [comment]
> 1. What to require for "Instances of [...]" dataset self-description?


Given our current minimum requirements, the following self-description triples would be required: 
| By DataCite:  | By UWLSWD: |
|---|---|
|- dct:title|- dc:language |
|- dct:creator OR dc:creator|- dct:license |
|- dct:publisher OR dc:publisher|- dct:description |
|- dct:issued|- dct:format | 
|- schema:disambiguatingDescription| - dct:hasFormat (can be added during serialization)|
||- schema:version|

- title, description, and date issued can just be moved from the VoID description dataset. 
- some resources have a creator, publisher, and/or license associated in the VoID dataset, but others will need these added
- dct:format, schema:disambiguatingDescription, and dc:language will need to be added 
- owl:version will need to be changed to schema:version (and we may want to update the version numbers)
- existing dct:hasFormat triples shouldn't be copied, ones with the updated URL are added during serialization

**Reply by briesenberg07** [reply]
This seems like a reasonable starting point! I'm updating [instances_of.csv]() based on your helpful summary @cspayne .

Unless I've completely misinterpreted @gerontakos above there is currently no disagreement about adding triples along these lines to each of the "Instances of (...)" datasets. Thus, let us move from this 'discussion' (initiated in **[1. What to require for "Instances of [...]" dataset self-description?](https://github.com/uwlib-cams/uwlswd/discussions/42#discussion-5555827)** above) to issue #43 , and decide on specific triples to add (or potentially take away) from the "Instances of (...)" resources!?

☝🏻 The *other* discussion (**[2. Whether/how to maintain/deprecate the uwlib.55.a dataset description resource and the uwlib.55.a#uwSemWeb dataset resource]()** above) / issue( #44 ) remains TBD. 

Also I'm reflecting on how I've created a discussion *and* two issues when I could've just created two issues and I apologize for this.
